### DISCUSSION TITLE
tls vs. https in reverse mode

### DISCUSSION THREAD

**Discussion by Lingxi-Li** [mitmproxy/mitmproxy#6116] [Q&A] [answered]
Consider `reverse:tls://example.com:443` vs. `reverse:https://example.com`. Why do we need to distinguish the two? Would mitmproxy behave any differently under the hood with 'tls' and 'https'? I've read the [doc](https://docs.mitmproxy.org/stable/concepts-modes/#reverse-proxy), but it does not help much.

**Comment by mhils** [comment, marked as answer]
Starting with mitmproxy 10:

1. `reverse:https://example.com` can be used to forward HTTP traffic only. This means mitmproxy will parse incoming messages as HTTP, and then trigger the usual `request`/`response`/... event hooks.
1. `tls://` can be used to forward raw TCP (over TLS). This means mitmproxy will not try to parse them as HTTP, and will only fire the `tcp_message` hook for transmitted data. This can be used to proxy protocols other than HTTP that are still protected using TLS.

The behavior with mitmproxy 9 and before is a bit different, it's basically the same there.
